tests/platform_tests/api/test_psu.py

```diff
@@ -204,68 +215,52 @@ def test_power(self, duthosts, enum_rand_one_per_hwsku_hostname, localhost, plat
-
-                powergood_status = psu.get_powergood_status(platform_api_conn, psu_id)
-                if self.expect(powergood_status is not None,
-                               "Failed to retrieve operational status of PSU {}".format(psu_id)):
-                    self.expect(powergood_status is True, "PSU {} is not operational".format(psu_id))
-
-                high_threshold = None
-                voltage_high_threshold_supported = self.get_psu_facts(duthost, psu_id, True, "voltage_high_threshold")
-                if voltage_high_threshold_supported:
-                    high_threshold = psu.get_voltage_high_threshold(platform_api_conn, psu_id)
-                    if self.expect(high_threshold is not None,
-                                   "Failed to retrieve the high voltage threshold of PSU {}".format(psu_id)):
-                        self.expect(isinstance(high_threshold, float),
-                                    "PSU {} voltage high threshold appears incorrect".format(psu_id))
-                low_threshold = None
-                voltage_low_threshold_supported = self.get_psu_facts(duthost, psu_id, True, "voltage_low_threshold")
-                if voltage_low_threshold_supported:
-                    low_threshold = psu.get_voltage_low_threshold(platform_api_conn, psu_id)
-                    if self.expect(low_threshold is not None,
-                                   "Failed to retrieve the low voltage threshold of PSU {}".format(psu_id)):
-                        self.expect(isinstance(low_threshold, float),
-                                    "PSU {} voltage low threshold appears incorrect".format(psu_id))
-                if high_threshold is not None and low_threshold is not None:
-                    self.expect(voltage < high_threshold and voltage > low_threshold,
-                                "Voltage {} of PSU {} is not in between {} and {}"
-                                .format(voltage, psu_id, low_threshold, high_threshold))
+            failure_count = self.get_len_failed_expectations()
+            failure_occured = False
+            psu_info = {
+                "duthost": duthost,
+                "api": platform_api_conn,
+                "psu_id": psu_id
+            }
+            for i in range(0, MAX_ATTEMPTS):
+                name = psu.get_name(platform_api_conn, psu_id)
+                if name in self.psu_skip_list:
+                    logger.info("skipping check for {}".format(name))
+                else:
+                    voltage = self.get_psu_parameter(psu_info, "voltage", psu.get_voltage, "voltage")
+                    current = self.get_psu_parameter(psu_info, "current", psu.get_current, "current")
+                    power = self.get_psu_parameter(psu_info, "power", psu.get_power, "power")
+
+                    failure_occured = self.get_len_failed_expectations() > failure_count
+                    if current is not None and voltage is not None and power is not None:
```

Comment:
'if current is not None and voltage is not None and power is not None: ' is changed to  **if current  and voltage and power :**
Maybe It is more concise?

Reply:
Resolved in the latest commit.